packages/liveblocks-core/src/client.ts

```diff
@@ -600,11 +623,136 @@ export function createClient<U extends BaseUserMeta = DU>(
+      },
+    });
+  }
+
+  async function getRoomThreads<M extends BaseMetadata>(
+    roomId: string,
+    options?: GetThreadsOptions<M>
+  ) {
+    let query: string | undefined;
+    if (options !== undefined && options.query !== undefined) {
+      query = objectToQuery(options.query);
+    }
+
+    const response = await fetchRoomApi(
+      roomId,
+      url`/v2/c/rooms/${roomId}/threads`,
+      { query },
+      { headers: { "Content-Type": "application/json" } }
+    );
+
+    if (response.ok) {
+      const json = await (response.json() as Promise<{
+        data: ThreadDataPlain<M>[];
+        inboxNotifications: InboxNotificationDataPlain[];
+        deletedThreads: ThreadDeleteInfoPlain[];
+        deletedInboxNotifications: InboxNotificationDeleteInfoPlain[];
+        meta: {
+          requestedAt: string;
+        };
+      }>);
+
+      return {
+        threads: json.data.map(convertToThreadData),
+        inboxNotifications: json.inboxNotifications.map(
+          convertToInboxNotificationData
+        ),
+        requestedAt: new Date(json.meta.requestedAt),
+      };
+    } else if (response.status === 404) {
+      return {
+        threads: [],
+        inboxNotifications: [],
+        deletedThreads: [],
+        deletedInboxNotifications: [],
+        requestedAt: new Date(),
+      };
+    } else {
+      throw new Error("There was an error while getting threads.");
+    }
+  }
+
+  async function getRoomThreadsSince<M extends BaseMetadata>(
+    roomId: string,
+    options: { since: Date }
+  ) {
+    const response = await fetchRoomApi(
+      roomId,
+      url`/v2/c/rooms/${roomId}/threads`,
+      { since: options?.since?.toISOString() },
+      { headers: { "Content-Type": "application/json" } }
+    );
+
+    if (response.ok) {
+      const json = await (response.json() as Promise<{
+        data: ThreadDataPlain<M>[];
+        inboxNotifications: InboxNotificationDataPlain[];
+        deletedThreads: ThreadDeleteInfoPlain[];
+        deletedInboxNotifications: InboxNotificationDeleteInfoPlain[];
+        meta: {
+          requestedAt: string;
+        };
+      }>);
+
+      return {
+        threads: {
+          updated: json.data.map(convertToThreadData),
+          deleted: json.deletedThreads.map(convertToThreadDeleteInfo),
+        },
+        inboxNotifications: {
+          updated: json.inboxNotifications.map(convertToInboxNotificationData),
+          deleted: json.deletedInboxNotifications.map(
+            convertToInboxNotificationDeleteInfo
+          ),
+        },
+        requestedAt: new Date(json.meta.requestedAt),
+      };
+    } else if (response.status === 404) {
+      return {
+        threads: {
+          updated: [],
+          deleted: [],
+        },
+        inboxNotifications: {
+          updated: [],
+          deleted: [],
+        },
+        requestedAt: new Date(),
+      };
+    } else {
+      throw new Error("There was an error while getting threads.");
+    }
+  }
```

Comment:
@nimeshnayaju For inbox notifications, I just DRYed these up, and only have a single getter, where `?since=` is now optional. The only difference is that if `since` is undefined, it will do an initial fetch (i.e. page 1), and otherwise it will do delta updates (used by the poller). Also, if `since` is present, there can be "deleted" threads. If `since` isn't set, it will always return an empty array of deleted threads. (This is at least how it works for inbox notifications in the backend.)

Reply:
> Also, if since is present, there can be "deleted" threads. If since isn't set, it will always return an empty array of deleted threads. (This is at least how it works for inbox notifications in the backend.)

I checked the backend for the `GET: threads` endpoint and it seems that the endpoint **CAN** return deleted threads even without `since` parameter. So, the batch of `2000` threads is deleted threads + non-deleted threads 🤔.

Reply:
It seems to me that that is not correct and should get fixed. Maybe I'm overlooking something, but there should be 2 use cases only I think:

1. The first use case is a "delta update", aka a `?since=` query. It should return _all_ (not capped to 2000, but unlimited) threads created since the given date, including all threads deleted since that moment.
2. The second use case is the initial fetch (= page 1) and all subsequent page fetches. Those will use `?cursor&limit` params (and never `?since=`) and should _not_ return any deleted threads, ever.

This is my current understanding.